tests/Microsoft.Identity.Web.Test/TokenAcquisitionAddInTests.cs

```diff
@@ -0,0 +1,74 @@
+using Microsoft.Identity.Abstractions;
+using Microsoft.Identity.Client;
+using NSubstitute;
+using Xunit;
+
+namespace Microsoft.Identity.Web.Tests
+{
+    public class TokenAcquisitionAddInTests
+    {
+        [Fact]
+        public void InvokeOnBuildConfidentialClientApplication_InvokesEvent()
```

Comment:
I think a "more end to end" test can be written. Smth like: 

- define a simple test AddIn which uses all 3 events
- call the TokenAcqusition `Acquire` methods
- Assert that the callbacks are invoked.

Happy to help write this.